__init__.py

```diff
@@ -558,35 +633,57 @@ def play_playlist_on(self, message):
                     self.spotify.transfer_playback(dev["id"], False)
                 self.start_playback(dev, playlist)
 
-    @intent_file_handler('PlaySpotify.intent')
-    def continue_current_playlist(self, message):
-        if self.playback_prerequisits_ok():
-            dev = self.get_default_device()
-            if dev:
-                self.spotify_play(dev['id'])
-            else:
-                self.speak_dialog('NoDevicesAvailable')
+#    @intent_file_handler('PlaySpotify.intent')
+#    def continue_current_playlist(self, message):
+#        if self.playback_prerequisits_ok():
+#            dev = self.get_default_device()
+#            if dev:
+#                self.spotify_play(dev['id'])
+#            else:
+#                self.speak_dialog('NoDevicesAvailable')
 
-    @intent_handler(IntentBuilder('').require('Search').
-                    require('AlbumTitle').require('Spotify'))
-    def search_album(self, message):
-        if self.playback_prerequisits_ok():
-            return self.search(message.data['AlbumTitle'], 'album')
-
-    def play_album(self, message):
-        if self.playback_prerequisits_ok():
-            return self.search(message.data['AlbumTitle'], 'album')
+    def play(self, data, type='track', genreName=None):
+        """
 
-    def play_artist(self, message):
-        if self.playback_prerequisits_ok():
-            return self.search(message.data['Artist'], 'artist')
+        :param data: data returned by self.search_spotify
+        :param type: the type of data. 'track', 'album', or 'genre' are currently supported
+        :param genreName: if type is 'genre', also include the genre's name here, for output purposes
+        """
+        dev = self.get_default_device()
+        if dev is None:
+            LOG.error("Unable to get a default device while trying to play something.")
+            self.speak_dialog('NoDevicesAvailable')
+        else:
+            try:
+                if type is 'track':
+                    song = data['tracks']['items'][0]
+                    self.speak_dialog('listening_to_song_by', data={'tracks': song['name'], 'artist': song['artists'][0]['name']})
+                    time.sleep(2)
+                    self.spotify_play(dev['id'], uris=[song['uri']])
+                elif type is 'album':
+                    album = data['albums']['items'][0]
+                    self.speak_dialog('listening_to_album_by', data={'album': album['name'], 'artist': album['artists'][0]['name']})
+                    self.speak_dialog('listening_to_album_by', data={'album': album['name'], 'artist': album['artists'][0]['name']})
```

Comment:
Duplicate.

We might also make it less verbose, just confirming the album maybe.

Reply:
Good catch, thanks. I made these changes, though I still have it speaking the album name by artist--I tried to make it shorter by removing "the album" portion though.

I included the artist information because I ran into situations where it selected an album I didn't expect at first, leading me to make an utterance clarifying who the artist was. If in testing this is still too long I could remove the artist too